src/inmanta/agent/forking_executor.py

```diff
@@ -390,6 +390,14 @@ async def join(self, timeout: float) -> None:
             return
         try:
             await asyncio.get_running_loop().run_in_executor(None, functools.partial(self.process.join, timeout))
+            if self.process.exitcode is None:
+                LOGGER.warning(
+                    "Executor for agent %s with id %s didn't stop after timeout of %d seconds. Killing it.",
+                    self.executor_id.agent_name,
+                    self.executor_id.identity(),
+                    timeout,
+                )
+                self.process.kill()
```

Comment:
Don't we need to join after kill to prevent a zombie process?

Reply:
The documentation is rather vague on this, but it doesn't hurt to do the join.